src/ray/gcs/gcs_server/gcs_node_manager.cc

```diff
@@ -77,7 +79,12 @@ void GcsNodeManager::NodeFailureDetector::SendBatchedHeartbeat() {
     for (const auto &heartbeat : heartbeat_buffer_) {
       batch->add_batch()->CopyFrom(heartbeat.second);
     }
-    RAY_CHECK_OK(node_info_accessor_.AsyncReportBatchHeartbeat(batch, nullptr));
+
+    auto done = [this, batch](Status status) {
```

Comment:
I think `node_info_accessor_.AsyncReportBatchHeartbeat(...)` could be removed as we just want to publish the BatchHeartbeat to nodes. You could invoke `gcs_pub_sub_->Publish(...)` directly instead of `node_info_accessor_.AsyncReportBatchHeartbeat(...)`;

Reply:
fixed

Reply:
test_multi_node_2.py subscribes HEARTBEAT_BATCH_CHANNEL, so we can not remove it.